dataGenerator.js

```diff
@@ -23,149 +23,939 @@ var db = require('./server/db.js');
+ */
+var answerQuestionsRegular = function () {
+    for (var id = questionsIds; id < questionsCount; id++) {
+        answerQuestionRegular(id);
+    }
+}
+
+/**
+ * This function answers the regular type question
+ * 
+ * @param {integer} questionId 
+ */
+var answerQuestionRegular = function (questionId) {
     for (var i = 0; i < questionsAttempts; i++) {
-        var studentId = 'student'+Math.floor(Math.random()*studentsCount);
+        var studentId = allIds[Math.floor(Math.random() * studentsCount)];
         var answer = 'NotKonniChiwa';
 
-        if (Math.floor(Math.random()*100) > (100-questionsCorrectPercentage)) {
+        if (Math.floor(Math.random() * 100) > (100 - questionsCorrectPercentage)) {
             answer = 'KonniChiwa';
         }
 
         checkAnswer(questionId, studentId, answer, function (err, correct) {
             if (err) {
                 logger.error(err);
             } else {
-                logger.info('Questions %d answered %s by %s', questionId, correct? 'correctly' : 'incorrectly', studentId);
+                logger.info('Questions %d answered %s by %s', questionId, correct ? 'correctly' : 'incorrectly', studentId);
+            }
+
+            if (questionsAnswered === numberOfQuestionsExpected * questionsAttempts) {
+                createQuestionsMultipleChoice();
             }
 
             questionsAnswered++;
-            if (questionsAnswered == questionsCount*questionsAttempts) {
-                process.exit(0);
+        });
+    }
+}
+
+/**
+ * This function creates all the multiple choice type question based on the variables
+ */
+var createQuestionsMultipleChoice = function () {
+    variableReset('multipleChoice');
+    for (var id = questionsIds; id < questionsCount; id++) {
+        addQuestionMultipleChoice('Is math related to science? ' + id, id);
+    }
+}
+
+/**
+ * This function adds the multiple choice type question
+ * 
+ * @param {string} qTopic 
+ * @param {integer} id 
+ */
+var addQuestionMultipleChoice = function (qTopic, id) {
+    var question = {
+        topic: 'CSC492',
+        title: qTopic,
+        text: '<p>' + qTopic + ' Text</p>',
+        answer: 'Option1',
+        points: Math.floor(Math.random() * questionsMaxValue),
+        type: common.questionTypes.MULTIPLECHOICE.value,
+        hint: 'Option1',
+        visible: 'true',
+        choices: ['Option1', 'Option2', 'Option3', 'Option4']
+    };
+
+    questions.addQuestion(question, function (err, res) {
+        if (err) {
+            logger.error('Could not add question. Please try again.');
+        } else {
+            logger.info('Questions %d created', id);
+
+            for (var i = 0; i < adminsCount; i++) {
+                rateQuestion(id, 'Admin' + i, Math.floor(Math.random() * 6));
+            }
+        }
+
+        if (questionsCreated === numberOfQuestionsExpected) {
+            answerQuestionsMultipleChoice();
+        }
+
+        questionsCreated++;
```

Comment:
since the counts start at 0, should we check the if statement after the counting?

Reply:
that is how it used to be previously, and there is a delayed call that used to run after that if statement, this fixes that issue